pallets/staking-rewards/src/lib.rs

```diff
@@ -367,19 +367,22 @@ impl<T: Config> OnUnbalanced<PositiveImbalanceOf<T>> for Pallet<T> {
     fn on_nonzero_unbalanced(minted: PositiveImbalanceOf<T>) {
         let amount = minted.peek();
 
-        let burned = T::Currency::withdraw(
+        if let Ok(burned) = T::Currency::withdraw(
             &Self::account_id(),
             amount,
             WithdrawReasons::TRANSFER,
             ExistenceRequirement::KeepAlive,
```

Comment:
are you sure we need to keep alive?
for cases of big withdrawals this may fail each time because resulting balance will go lower than ED but with keep alive we cannot kill it, so error will be returned each time, burn won't take place while total supply increases what leads to invalidation of tokenomic model - better to have a test for it

Reply:
> for cases of big withdrawals

This should never be the case. The purpose of this account is to store the inflation offset pool. It is never supposed to be completely depleted: one-time withdrawal can only be as big as a single era reward for one validator, whereas the pool in the beginning is supposed to store 10% of the total token supply, that is a value of totally different order of magnitude which never gets even close to the ED. According to the tokenomics model the pool goes out of the picture in a year or two anyway.